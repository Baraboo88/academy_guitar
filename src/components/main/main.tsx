import React, {useCallback, useEffect, useState} from 'react';
import {getDataError, getDataIsResponseReceived, getGuitars} from '../../store/data/data-selectors';
import {StateModel} from '../../types/redux-models';

import { DataOperation} from '../../store/data/data-reducer';
import {connect} from 'react-redux';
import {GuitarModel} from '../../types/guitar-model';
import GuitarCard from '../guitar-card/guitar-card';
import {Link, RouteComponentProps} from 'react-router-dom';
import Header from '../header/header';
import Footer from '../footer/footer';

const ITEMS_ON_THE_PAGE = 9;


interface MatchParams {
  id?: string;
}

interface MainProps {
  guitars: GuitarModel [];
  onMount: () => void;
  isResponseReceived: boolean;
  errorMsg: string;
  getCommentsCount: (guitars: GuitarModel []) => void;
}


function Main(props: MainProps & RouteComponentProps<MatchParams>) {
  const {onMount, guitars, isResponseReceived, errorMsg, getCommentsCount, history} = props;
  const [error, setError] = useState<string>('');
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [innerGuitars, setInnerGuitars] = useState<GuitarModel []>([]);
  const [currentPage, setCurrentPage] = useState(1);
  const [isLoadingCommentsCount, setIsLoadingCommentsCount] = useState<boolean>(true);

  useEffect(() => {
    if (isResponseReceived && errorMsg) {
      setError(errorMsg);
    }
    if (!isResponseReceived && guitars.length === 0) {
      onMount();
    } else {
      setIsLoading(false);
      setInnerGuitars(guitars);
      if(isLoadingCommentsCount && guitars.length > 0){
        getCommentsCount(guitars);
        setIsLoadingCommentsCount(false);
      }
    }

  }, [guitars, onMount, isResponseReceived, errorMsg, getCommentsCount, isLoadingCommentsCount]);


  const getAllPages = useCallback(() => {
    let allPages = 0;
    if (guitars) {
      if (guitars.length % ITEMS_ON_THE_PAGE === 0) {
        allPages = Math.floor(guitars.length / ITEMS_ON_THE_PAGE);
      } else {
        allPages = Math.floor(guitars.length / ITEMS_ON_THE_PAGE) + 1;
      }
    }
    return allPages;
  }, [guitars]);

  useEffect(() => {
    if(props.match.params.id){
      setCurrentPage(Number(props.match.params.id));
    } else {
      setCurrentPage(1);
    }

    if(guitars && guitars.length > 0 && getAllPages() < Number(props.match.params.id)){
      history.push('/not-found');
    }

  }, [props.match.params.id, guitars, getAllPages, history]);


  const renderPagination = () => {


    const allPages = getAllPages();
    const pages = [];
    for (let i = 1; i <= allPages; i++) {
      pages.push(i);
    }
    return pages.map((page) =>(
      <li className={`pagination__page ${page === currentPage ? 'pagination__page--active' : ''}`} key={page}>
        <Link to={`/catalog/page/${page}`} className="link pagination__page-link" href="#">
          {page}
        </Link>
      </li>));
  };


  return (
    <div className="wrapper">
      <Header/>
      <main className="page-content">
        <div className="container">
          <h1 className="page-content__title title title--bigger">Каталог гитар</h1>
          <ul className="breadcrumbs page-content__breadcrumbs">
            <li className="breadcrumbs__item"><Link to={'/'} className="link">Главная</Link>
            </li>
            <li className="breadcrumbs__item"><span >Каталог</span>
            </li>
          </ul>
          <div className="catalog">
            <form className="catalog-filter">
              <h2 className="title title--bigger catalog-filter__title">Фильтр</h2>
              <fieldset className="catalog-filter__block">
                <legend className="catalog-filter__block-title">Цена, ₽</legend>
                <div className="catalog-filter__price-range">
                  <div className="form-input">
                    <label className="visually-hidden">Минимальная цена</label>
                    <input type="number" placeholder="1 000" id="priceMin" name="от"/>
                  </div>
                  <div className="form-input">
                    <label className="visually-hidden">Максимальная цена</label>
                    <input type="number" placeholder="30 000" id="priceMax" name="до"/>
                  </div>
                </div>
              </fieldset>
              <fieldset className="catalog-filter__block">
                <legend className="catalog-filter__block-title">Тип гитар</legend>
                <div className="form-checkbox catalog-filter__block-item">
                  <input className="visually-hidden" type="checkbox" id="acoustic" name="acoustic"/>
                  <label htmlFor="acoustic">Акустические гитары</label>
                </div>
                <div className="form-checkbox catalog-filter__block-item">
                  <input defaultChecked className="visually-hidden" type="checkbox" id="electric" name="electric"/>
                  <label htmlFor="electric">Электрогитары</label>
                </div>
                <div className="form-checkbox catalog-filter__block-item">
                  <input defaultChecked className="visually-hidden" type="checkbox" id="ukulele" name="ukulele"/>
                  <label htmlFor="ukulele">Укулеле</label>
                </div>
              </fieldset>
              <fieldset className="catalog-filter__block">
                <legend className="catalog-filter__block-title">Количество струн</legend>
                <div className="form-checkbox catalog-filter__block-item">
                  <input defaultChecked className="visually-hidden" type="checkbox" id="4-strings" name="4-strings"/>
                  <label htmlFor="4-strings">4</label>
                </div>
                <div className="form-checkbox catalog-filter__block-item">
                  <input defaultChecked className="visually-hidden" type="checkbox" id="6-strings" name="6-strings"/>
                  <label htmlFor="6-strings">6</label>
                </div>
                <div className="form-checkbox catalog-filter__block-item">
                  <input className="visually-hidden" type="checkbox" id="7-strings" name="7-strings"/>
                  <label htmlFor="7-strings">7</label>
                </div>
                <div className="form-checkbox catalog-filter__block-item">
                  <input disabled className="visually-hidden" type="checkbox" id="12-strings" name="12-strings"/>
                  <label htmlFor="12-strings">12</label>
                </div>
              </fieldset>
              <button className="catalog-filter__reset-btn button button--black-border button--medium"
                type="reset"
              >
                                Очистить
              </button>
            </form>
            <div className="catalog-sort">
              <h2 className="catalog-sort__title">Сортировать:</h2>
              <div className="catalog-sort__type">
                <button className="catalog-sort__type-button" aria-label="по цене">по цене</button>
                <button className="catalog-sort__type-button" aria-label="по популярности">по
                                    популярности
                </button>
              </div>
              <div className="catalog-sort__order">
                <button className="catalog-sort__order-button catalog-sort__order-button--up"
                  aria-label="По возрастанию"
                >
                </button>
                <button className="catalog-sort__order-button catalog-sort__order-button--down"
                  aria-label="По убыванию"
                >
                </button>
              </div>
            </div>
            <div className="cards catalog__cards">
              {error && <span style={{color: 'red', textAlign: 'center'}}>Something went wrong</span>}
              {!isLoading && innerGuitars.slice((currentPage - 1) * ITEMS_ON_THE_PAGE,  (currentPage - 1) * ITEMS_ON_THE_PAGE + ITEMS_ON_THE_PAGE).map((guitar) => <GuitarCard key={guitar.id} card={guitar}/>)}
            </div>
            <div className="pagination page-content__pagination">
              <ul className="pagination__list">
                {currentPage > 1 ?
                  <li className="pagination__page pagination__page--prev" id="prev">
                    <Link to={`/catalog/page/${currentPage - 1}`} className="link pagination__page-link">
                      Назад
                    </Link>
                  </li> : ''}
                {renderPagination()}
                { getAllPages() !== 0 && currentPage < getAllPages()  ?
                  <li className="pagination__page pagination__page--next" id="next">
                    <Link to={`/catalog/page/${currentPage + 1}`} className="link pagination__page-link">
                     Далее
                    </Link>
                  </li> : ''}
              </ul>
            </div>
          </div>
        </div>
      </main>
      <Footer/>
    </div>
  );
}


const mapStateToProps = (state: StateModel) => ({
  guitars: getGuitars(state),
  isResponseReceived: getDataIsResponseReceived(state),
  errorMsg: getDataError(state),
});

/* eslint-disable  @typescript-eslint/no-explicit-any */
const mapDispatchToProps = (dispatch: any) => ({
  onMount() {
    dispatch(DataOperation.getGuitars());
  },
  getCommentsCount(guitars: GuitarModel []){
    dispatch(DataOperation.getCommentsCount(guitars));
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(Main);

export {Main};
