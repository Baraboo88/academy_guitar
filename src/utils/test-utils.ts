import {AddCommentModel, GuitarCommentModel, GuitarModel, GuitarType} from '../types/guitar-model';
import {createApi} from '../api';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {ReactWrapper} from 'enzyme';


export const MOCK_ERROR= 'Somme error';

export const mockGuitars: GuitarModel[] = [
  {
    id: 1,
    name: 'Честер Bass',
    vendorCode: 'SO757575',
    type: 'electric' as GuitarType,
    description: 'Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 4,
    price: 17500,
  },
  {
    id: 2,
    name: 'CURT Z300',
    vendorCode: 'TK129049',
    type: 'electric' as GuitarType,
    description: 'Эргономичность гитары и качество сборки являются, пожалуй, её главными преимуществами. Идеальное расположение в руках музыканта дополняется прочностью конструкции из клёна.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 29500,
  },
  {
    id: 3,
    name: 'Roman LX',
    vendorCode: 'RO111111',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле класса премиум от компании CURT, собравшая в себе все самые необходимые качесва: лёгкость корпуса, прочность струн и компактный размер.',
    previewImg: 'img/guitar-6.jpg',
    stringCount: 4,
    rating: 4,
    price: 6800,
  },
  {
    id: 4,
    name: 'CURT T300',
    vendorCode: 'TK436457',
    type: 'electric' as GuitarType,
    description: 'CURT T300 - это шестиструнная электрогитара популярной линейки FPT. Модель с классическим стилем головы грифа и деки.',
    previewImg: 'img/guitar-3.jpg',
    stringCount: 6,
    rating: 5,
    price: 30000,
  },
  {
    id: 5,
    name: 'Dania Super',
    vendorCode: 'DI192138',
    type: 'acoustic' as GuitarType,
    description: 'Гитары производителя Dania пользуются популярностью у музыкантов разного уровня. В модели Super идеально сочетаются демократичная цена и качество. Корпус, выполненный из тополя позволяет корпусу долгое время сохранять свой первоначальный вид.',
    previewImg: 'img/guitar-4.jpg',
    stringCount: 7,
    rating: 4.5,
    price: 3500,
  },
  {
    id: 6,
    name: 'Честер WX',
    vendorCode: 'SO934345',
    type: 'electric' as GuitarType,
    description: 'Электрогитара с олд-скульным грифом и глянцевым корпусом для настоящих рок-музыкантов. Насыщенный звук идеально подойдет для исполнения рок-композиций.',
    previewImg: 'img/guitar-2.jpg',
    stringCount: 6,
    rating: 3.5,
    price: 15300,
  },
  {
    id: 7,
    name: 'Dania VX',
    vendorCode: 'DI082347',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле - прекрасный компактный аналог классической акустической гитары. Прекрасное качество продуктов компании Dania сочетается с демократичными ценами продуктов линейки GSR.',
    previewImg: 'img/guitar-5.jpg',
    stringCount: 4,
    rating: 2.5,
    price: 2200,
  },
];
export const testInitialState = {
  currentGuitar: {
    currentGuitar: mockGuitars,
    errorMsg: '',
  },
  guitars: {
    guitars: mockGuitars,
    errorMsg: '',
  },
};

export const mockCommentToAdd: AddCommentModel = {
  advantage: 'Mock Advantage', comment: 'Mock Comment', disadvantage: 'Mock Disadvantage', guitarId: 3, rating: 3, userName: 'Some name',
};


export const mockComment: GuitarCommentModel = {id: '31b10247-adff-44a7-937e-c71af64f425b',
  userName: 'Никита',
  advantage: 'Цена.',
  disadvantage: 'Быстро приходит в негодность покрытие краской.',
  comment: 'Хорошая гитара для начинающих, сделана качественно. Лучше за эти деньги не найти.',
  rating: 5,
  createAt: new Date('2022-03-10T10:56:01.590Z'),
  guitarId: 1};

export const getTestStore = () => {

  const reducer = (state = testInitialState) => state;
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};

export const findByTestAtr = (component:ReactWrapper, attr: string) => component.find(`[data-test='${attr}']`);
