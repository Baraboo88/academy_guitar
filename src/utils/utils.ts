const STARTS_AMOUNT = 5;

export const imageAdapter = (img: string) => img.slice(img.length - 5, img.length - 4);


export const generateStarsArray = () => {
  const stars = [];
  for (let i = 1; i <= STARTS_AMOUNT; i++) {
    stars.push(i);
  }
  return stars;
};

export const getCyrillicRating = (rating: number) => {
  if(rating> 0 && rating < 3){
    return 'Плохо';
  } else if (rating >= 3 && rating < 4){
    return 'Удовлетворительно';
  } else if(rating >= 4 && rating < 5){
    return 'Хорошо';
  } else {
    return 'Отлично';
  }
};
