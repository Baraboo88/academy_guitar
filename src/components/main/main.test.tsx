import * as Enzyme from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {getTestStore, mockGuitars} from '../../utils/test-utils';
import {BrowserRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json';
import {Main} from './main';
import {SortDirection, SortType} from '../../utils/utils';

Enzyme.configure({adapter: new EnzymeReactAdapter()});


it('Main successfully rendered', () => {

  const mockGetCommentsCount = jest.fn();

  const MOCK_ERROR = '';


  const tree = mount(
    <Provider store={getTestStore()}>
      <BrowserRouter>
        <Main
          guitarsWithoutSort = {mockGuitars}
          guitars = {mockGuitars}
          getCommentsCount = {mockGetCommentsCount}
          isResponseReceived ={false}
          errorMsg={MOCK_ERROR}
          sortDirection = {SortDirection.None}
          sortType={SortType.Price}
          setSortType={jest.fn()}
          setSortDirection={jest.fn()}
        />
      </BrowserRouter>
    </Provider>,
  );
  expect(toJson(tree, {mode: 'deep'})).toMatchSnapshot();
});
