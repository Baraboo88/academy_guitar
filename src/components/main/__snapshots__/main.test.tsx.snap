// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main successfully rendered 1`] = `
<div
  className="wrapper"
>
  <header
    className="header"
    id="header"
  >
    <div
      className="container header__wrapper"
    >
      <a
        className="header__logo logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Логотип"
          className="logo__img"
          height="70"
          src="/img/svg/logo.svg"
          width="70"
        />
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li>
            <a
              className="link main-nav__link link--current"
              href="/"
              onClick={[Function]}
            >
              Каталог
            </a>
          </li>
          <li>
            <a
              className="link main-nav__link"
              href="#"
            >
              Где купить?
            </a>
          </li>
          <li>
            <a
              className="link main-nav__link"
              href="#"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="form-search"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <form
          className="form-search__form"
          id="form-search"
        >
          <button
            className="form-search__submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="form-search__icon"
              height="15"
              width="14"
            >
              <use
                xlinkHref="#icon-search"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              Начать поиск
            </span>
          </button>
          <input
            autoComplete="off"
            className="form-search__input"
            data-test="test-search-name"
            id="search"
            onChange={[Function]}
            placeholder="что вы ищите?"
            type="text"
          />
          <label
            className="visually-hidden"
            htmlFor="search"
          >
            Поиск
          </label>
        </form>
        <ul
          className="form-search__select-list hidden"
        >
          <li
            className="form-search__select-item"
            key="1"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Честер Bass
          </li>
          <li
            className="form-search__select-item"
            key="2"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            CURT Z300
          </li>
          <li
            className="form-search__select-item"
            key="3"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Roman LX
          </li>
          <li
            className="form-search__select-item"
            key="4"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            CURT T300
          </li>
          <li
            className="form-search__select-item"
            key="5"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Dania Super
          </li>
          <li
            className="form-search__select-item"
            key="6"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Честер WX
          </li>
          <li
            className="form-search__select-item"
            key="7"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Dania VX
          </li>
        </ul>
        <button
          className="form-search__reset"
          form="form-search"
          onClick={[Function]}
          type="reset"
        >
          <svg
            aria-hidden="true"
            className="form-search__icon"
            height="15"
            width="14"
          >
            <use
              xlinkHref="#icon-close"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            Сбросить поиск
          </span>
        </button>
      </div>
      <a
        aria-label="Корзина"
        className="header__cart-link"
        href="#"
      >
        <svg
          aria-hidden="true"
          className="header__cart-icon"
          height="14"
          width="14"
        >
          <use
            xlinkHref="#icon-basket"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          Перейти в корзину
        </span>
        <span
          className="header__cart-count"
        >
          2
        </span>
      </a>
    </div>
  </header>
  <main
    className="page-content"
  >
    <div
      className="container"
    >
      <h1
        className="page-content__title title title--bigger"
      >
        Каталог гитар
      </h1>
      <ul
        className="breadcrumbs page-content__breadcrumbs"
      >
        <li
          className="breadcrumbs__item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Главная
          </a>
        </li>
        <li
          className="breadcrumbs__item"
        >
          <span>
            Каталог
          </span>
        </li>
      </ul>
      <div
        className="catalog"
      >
        <form
          className="catalog-filter"
        >
          <h2
            className="title title--bigger catalog-filter__title"
          >
            Фильтр
          </h2>
          <fieldset
            className="catalog-filter__block"
          >
            <legend
              className="catalog-filter__block-title"
            >
              Цена, ₽
            </legend>
            <div
              className="catalog-filter__price-range"
            >
              <div
                className="form-input"
              >
                <label
                  className="visually-hidden"
                >
                  Минимальная цена
                </label>
                <input
                  id="priceMin"
                  name="от"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="2 200"
                  type="number"
                  value=""
                />
              </div>
              <div
                className="form-input"
              >
                <label
                  className="visually-hidden"
                >
                  Максимальная цена
                </label>
                <input
                  id="priceMax"
                  name="до"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="30 000"
                  type="number"
                  value=""
                />
              </div>
            </div>
          </fieldset>
          <fieldset
            className="catalog-filter__block"
          >
            <legend
              className="catalog-filter__block-title"
            >
              Тип гитар
            </legend>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="acoustic"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={false}
                id="acoustic"
                name="acoustic"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="acoustic"
              >
                Акустические гитары
              </label>
            </div>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="electric"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={false}
                id="electric"
                name="electric"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="electric"
              >
                Электрогитары
              </label>
            </div>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="ukulele"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={false}
                id="ukulele"
                name="ukulele"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="ukulele"
              >
                Укулеле
              </label>
            </div>
          </fieldset>
          <fieldset
            className="catalog-filter__block"
          >
            <legend
              className="catalog-filter__block-title"
            >
              Количество струн
            </legend>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="4"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={true}
                id="4-strings"
                name="4-strings"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="4-strings"
              >
                4
              </label>
            </div>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="6"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={true}
                id="6-strings"
                name="6-strings"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="6-strings"
              >
                6
              </label>
            </div>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="7"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={true}
                id="7-strings"
                name="7-strings"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="7-strings"
              >
                7
              </label>
            </div>
            <div
              className="form-checkbox catalog-filter__block-item"
              key="12"
            >
              <input
                checked={false}
                className="visually-hidden"
                disabled={true}
                id="12-strings"
                name="12-strings"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="12-strings"
              >
                12
              </label>
            </div>
          </fieldset>
          <button
            className="catalog-filter__reset-btn button button--black-border button--medium"
            onClick={[Function]}
            type="reset"
          >
            Очистить
          </button>
        </form>
        <div
          className="catalog-sort"
        >
          <h2
            className="catalog-sort__title"
          >
            Сортировать:
          </h2>
          <div
            className="catalog-sort__type"
          >
            <button
              aria-label="по цене"
              className="catalog-sort__type-button catalog-sort__type-button--active"
              onClick={[Function]}
            >
              по цене
            </button>
            <button
              aria-label="по популярности"
              className="catalog-sort__type-button "
              onClick={[Function]}
            >
              по популярности
            </button>
          </div>
          <div
            className="catalog-sort__order"
          >
            <button
              aria-label="По возрастанию"
              className="catalog-sort__order-button catalog-sort__order-button--up  "
              onClick={[Function]}
            />
            <button
              aria-label="По убыванию"
              className="catalog-sort__order-button  catalog-sort__order-button--down"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="cards catalog__cards"
        >
          <div
            style={
              Object {
                "gridColumn": 3,
                "justifySelf": "center",
              }
            }
          >
            <div
              data-testid="tail-spin-loading"
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <svg
                aria-label="tail-spin-loading"
                data-testid="tail-spin-svg"
                height={80}
                viewBox="0 0 38 38"
                width={80}
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <linearGradient
                    id="a"
                    x1="8.042%"
                    x2="65.682%"
                    y1="0%"
                    y2="23.865%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#000000"
                      stopOpacity="0"
                    />
                    <stop
                      offset="63.146%"
                      stopColor="#000000"
                      stopOpacity=".631"
                    />
                    <stop
                      offset="100%"
                      stopColor="#000000"
                    />
                  </linearGradient>
                </defs>
                <g
                  fill="none"
                  fillRule="evenodd"
                >
                  <g
                    transform="translate(1 1)"
                  >
                    <path
                      d="M36 18c0-9.94-8.06-18-18-18"
                      id="Oval-2"
                      stroke="#000000"
                      strokeWidth="2"
                    >
                      <animateTransform
                        attributeName="transform"
                        dur="0.9s"
                        from="0 18 18"
                        repeatCount="indefinite"
                        to="360 18 18"
                        type="rotate"
                      />
                    </path>
                    <circle
                      cx="36"
                      cy="18"
                      fill="#fff"
                      r={1}
                    >
                      <animateTransform
                        attributeName="transform"
                        dur="0.9s"
                        from="0 18 18"
                        repeatCount="indefinite"
                        to="360 18 18"
                        type="rotate"
                      />
                    </circle>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          className="pagination page-content__pagination"
        >
          <ul
            className="pagination__list"
          >
            <li
              className="pagination__page pagination__page--active"
              key="1"
            >
              <a
                className="link pagination__page-link"
                href="/?page=1"
                onClick={[Function]}
              >
                1
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <div
      className="footer__container container"
    >
      <a
        className="footer__logo logo"
        href="/"
      >
        <img
          alt="Логотип"
          className="logo__img"
          height="70"
          src="/img/svg/logo.svg"
          width="70"
        />
      </a>
      <div
        className="socials footer__socials"
      >
        <ul
          className="socials__list"
        >
          <li
            className="socials-item"
          >
            <a
              aria-label="skype"
              className="socials__link"
              href="https://www.skype.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-skype"
                />
              </svg>
            </a>
          </li>
          <li
            className="socials-item"
          >
            <a
              aria-label="vsco"
              className="socials__link"
              href="https://www.vsco.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-vsco"
                />
              </svg>
            </a>
          </li>
          <li
            className="socials-item"
          >
            <a
              aria-label="pinterest"
              className="socials__link"
              href="https://www.pinterest.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-pinterest"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <section
        className="footer__nav-section footer__nav-section--info"
      >
        <h2
          className="footer__nav-title"
        >
          О нас
        </h2>
        <p
          className="footer__nav-content footer__nav-content--font-secondary"
        >
          Магазин гитар, музыкальных инструментов и гитарная мастерская 
          <br />
           в Санкт-Петербурге.
          <br />
          <br />
          Все инструменты проверены, отстроены 
          <br />
           и доведены до идеала!
        </p>
      </section>
      <section
        className="footer__nav-section footer__nav-section--links"
      >
        <h2
          className="footer__nav-title"
        >
          Информация
        </h2>
        <ul
          className="footer__nav-list"
        >
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Где купить?
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Блог
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Вопрос - ответ
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Возврат
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Сервис-центры
            </a>
          </li>
        </ul>
      </section>
      <section
        className="footer__nav-section footer__nav-section--contacts"
      >
        <h2
          className="footer__nav-title"
        >
          Контакты
        </h2>
        <p
          className="footer__nav-content"
        >
          г. Санкт-Петербург,
          <br />
           м. Невский проспект, 
          <br />
          ул. Казанская 6.
        </p>
        <div
          className="footer__nav-content"
        >
          <svg
            aria-hidden="true"
            className="footer__icon"
            height="8"
            width="8"
          >
            <use
              xlinkHref="#icon-phone"
            />
          </svg>
          <a
            className="link"
            href="tel:88125005050"
          >
             8-812-500-50-50
          </a>
        </div>
        <p
          className="footer__nav-content"
        >
          Режим работы:
          <br />
          <span
            className="footer__span"
          >
            <svg
              aria-hidden="true"
              className="footer__icon"
              height="13"
              width="13"
            >
              <use
                xlinkHref="#icon-clock"
              />
            </svg>
            <span>
               с 11:00 до 20:00
            </span>
            <span>
              без выходных
            </span>
          </span>
        </p>
      </section>
    </div>
  </footer>
</div>
`;
